service/matching/taskListManager.go

```diff
@@ -476,12 +473,7 @@ func (c *taskListManagerImpl) executeWithRetry(
 
 	throttleRetry := backoff.NewThrottleRetry(
 		backoff.WithRetryPolicy(persistenceOperationRetryPolicy),
-		backoff.WithRetryableError(func(err error) bool {
-			if _, ok := err.(*persistence.ConditionFailedError); ok {
-				return false
-			}
-			return persistence.IsTransientError(err)
-		}),
+		backoff.WithRetryableError(persistence.IsTransientError),
```

Comment:
hm.. not related to this change, but we need to change persistence.IsTransientError to use errors.is/as